### PR TITLE
Support bare hostname in proxy

### LINKED ISSUE #1082: Should we support bare hostnames in `ALL_PROXY`,`HTTP_PROXY`, `HTTPS_PROXY`?
### Checklist

- [x] The bug is reproducible against the latest release and/or `master`.
- [x] There are no similar issues or pull requests to fix it yet.

### Describe the bug

I am a new user for httpx, and I just install it, then try to run it in python REPL.
I type `httpx.get('https://www.github.com')`  and then got the "No scheme included in URL".
So I try to debug it, and finally I found that because I set the `export ALL_PROXY=127.0.0.1:7890` in the shell, the proxy does not have a scheme. 
This proxy set is correct for when I use curl、brew and many other programe, So it should also work fun in httpx?
for any other reason httpx must have the scheme, I think it should prompt "a proxy set error" at least.

### Debugging material

```
~/projects » python3                                                                                                       hejl@hejldeIMAC
Python 3.8.4 (default, Jul 14 2020, 02:58:48) 
[Clang 11.0.3 (clang-1103.0.32.62)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import httpx
>>> httpx.get("https://www.github.com")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.8/site-packages/httpx/_api.py", line 159, in get
    return request(
  File "/usr/local/lib/python3.8/site-packages/httpx/_api.py", line 83, in request
    with Client(
  File "/usr/local/lib/python3.8/site-packages/httpx/_client.py", line 467, in __init__
    proxy_map = self.get_proxy_map(proxies, trust_env)
  File "/usr/local/lib/python3.8/site-packages/httpx/_client.py", line 87, in get_proxy_map
    return {
  File "/usr/local/lib/python3.8/site-packages/httpx/_client.py", line 88, in <dictcomp>
    key: Proxy(url=url)
  File "/usr/local/lib/python3.8/site-packages/httpx/_config.py", line 335, in __init__
    url = URL(url)
  File "/usr/local/lib/python3.8/site-packages/httpx/_models.py", line 90, in __init__
    raise InvalidURL("No scheme included in URL.")
httpx._exceptions.InvalidURL: No scheme included in URL.
```

### Environment

- OS:  MacOS
- Python version: 3.8.4
- HTTPX version: 0.13.3
- Async environment: no
- HTTP proxy: yes  `127.0.0.1:7890`
- Custom certificates: no

### PR DESCRIPTION
**Pull request by j178**
Closes #1082 

- Auto prepend `http` scheme for bare hostname in proxy
- Fix two internal use of deprcating `.getlist()`

### CODE REVIEW

File: tests/models/test_headers.py [hunk 1]
```diff
@@ -14,6 +14,7 @@ def test_headers():
     assert h.get("a") == "123, 456"
     assert h.get("nope", default=None) is None
     assert h.get_list("a") == ["123", "456"]
+    assert h.getlist("a") == ["123", "456"]
```
Comment (tomchristie) [inline tests/models/test_headers.py:17]:
```suggestion
```

File: tests/models/test_queryparams.py [hunk 2]
```diff
@@ -20,6 +20,7 @@ def test_queryparams(source):
     assert q.get("a") == "456"
     assert q.get("nope", default=None) is None
     assert q.get_list("a") == ["123", "456"]
+    assert q.getlist("a") == ["123", "456"]
```
Comment (tomchristie) [inline tests/models/test_queryparams.py:23]:
```suggestion
```

**Comment by tomchristie** [comment]
Fantastic, loving this, thanks! 😃

One interesting question here...
I *think* we probably only want to support this case wrt. bare environment variables, rather than supporting it in `Proxy` itself. (?)

Eg. If `get_environement_proxies` was changed from...

```python
for scheme in ("http", "https", "all"):
    if proxy_info.get(scheme):
        mounts[scheme] = proxy_info[scheme]
```

to instead be this...

```python
for scheme in ("http", "https", "all"):
    if proxy_info.get(scheme):
        hostname = proxy_info[scheme]
        mounts[scheme] = hostname if "://" in hostname else f"http://{hostname}"
```

Then we wouldn't change the code in `Proxy` at all, and our programmatic usage would remain a bit neater and more strict.

```python
client = httpx.Client(proxies="http://localhost:1234")  # Okay
client = httpx.Client(proxies="localhost:1234")         # Invalid, raises an error
```

While still handling the fuzzy `ALL_PROXY=localhost:1234` case gracefully.

Thoughts?

File: httpx/_models.py [hunk 3]
```diff
@@ -281,7 +281,7 @@ def update(self, params: QueryParamTypes = None) -> None:
 
         params = QueryParams(params)
         for param in params:
-            item, *extras = params.getlist(param)
+            item, *extras = params.get_list(param)
```
Comment (tomchristie) [inline httpx/_models.py:284]:
```suggestion
            item, *extras = params.getlist(param)
```

File: httpx/_models.py [hunk 4]
```diff
@@ -795,7 +795,7 @@ def decoder(self) -> Decoder:
         """
         if not hasattr(self, "_decoder"):
             decoders: typing.List[Decoder] = []
-            values = self.headers.getlist("content-encoding", split_commas=True)
+            values = self.headers.get_list("content-encoding", split_commas=True)
```
Comment (tomchristie) [inline httpx/_models.py:798]:
```suggestion
            values = self.headers.getlist("content-encoding", split_commas=True)
```

**Comment by florimondmanca** [comment]
Great stuff! Agreed with ^. :-) Also, if we go the "env vars only" path, we can probably update the [Environment Variables: HTTP_PROXY, HTTPS_PROXY, ALL_PROXY](https://www.python-httpx.org/environment_variables/#http_proxy-https_proxy-all_proxy) docs as part of this PR?

**Comment by tomchristie** [comment]
Great catch on the `get_list`/`getlist` - I've dropped those changes out of this PR.

We can treat them separately (Because I'd probably prefer that we `pragma: nocover` the to-be-deprecated lines, rather than include them in the test cases like that.)

**Comment by tomchristie** [comment]
@florimondmanca Yes, good call. General note: We're also missing `NO_PROXY` there.

**Comment by florimondmanca** [comment]
Yup. I'm working on a revamp of proxy docs anyway so I can include that stuff there as well.

**Pull request closed without merging**